### ISSUE TITLE
Native AOT

### ISSUE THREAD

**Issue by pshtiwan-hassan** [Finbuckle/Finbuckle.MultiTenant#874]
There could be issues with custom query filters in the MultiTenantDbContext when using AOT compilation with .NET 8, particularly in how the filters are applied or compiled. 

Is there any solution ?

**Comment by AndrewTriesToCode** [comment]
Yeah I believe with how I am using reflecting AOT compilation isn't a viable currently. I am working on some changes to how the query works though that should avoid this use of reflecting for most use cases.
